purchases/src/main/kotlin/com/revenuecat/purchases/attributes/SubscriberAttributesManager.kt

```diff
@@ -40,39 +41,72 @@ internal class SubscriberAttributesManager(
         setAttributes(mapOf(key.backendKey to value), appUserID)
     }
 
-    fun synchronizeSubscriberAttributesIfNeeded(
-        appUserID: String,
-        onSuccessHandler: () -> Unit,
-        onErrorHandler: (PurchasesError) -> Unit
+    fun synchronizeSubscriberAttributesForAllUsers(
+        currentAppUserID: AppUserID
     ) {
-        val unsyncedStoredAttributes = getUnsyncedSubscriberAttributes(appUserID)
+        val unsyncedStoredAttributesForAllUsers =
+            deviceCache.getUnsyncedSubscriberAttributes()
+        if (unsyncedStoredAttributesForAllUsers.isEmpty()) {
+            debugLog("No subscriber attributes to synchronize.")
+            return
+        }
 
-        if (unsyncedStoredAttributes.isNotEmpty()) {
-            debugLog("Synchronizing subscriber attributes: $unsyncedStoredAttributes")
+        unsyncedStoredAttributesForAllUsers.forEach { (appUserID, unsyncedAttributesForUser) ->
             backend.postSubscriberAttributes(
-                unsyncedStoredAttributes,
+                unsyncedAttributesForUser,
                 appUserID,
                 {
-                    markAsSynced(appUserID, unsyncedStoredAttributes, emptyList())
-                    onSuccessHandler()
+                    markAsSynced(appUserID, unsyncedAttributesForUser, emptyList())
+                    debugLog("Subscriber attributes synced successfully for appUserID: $appUserID.")
+                    if (currentAppUserID != appUserID) {
+                        deviceCache.clearSubscriberAttributesIfSynced(appUserID)
```

Comment:
There is an issue here.

1. User "cesar" syncs, we save it as synced.
2. User "pedro" syncs, we save it as synced. But we don't delete "cesar" synced attributes because it was appUserID is "pedro", we are not posting anything for "cesar".

I think the logic should be clear everything that is synced for the other appUserIds. I will update

Reply:
Updated in ea14999

Reply:
but doesn't this get called once per appUserID whose attributes were synced? 
that's what's happening on iOS at least. if that's the case, then for each appUserId we decide whether to delete attributes or not. I think this is the case on android as well.

Reply:
let me know if you wanna sync up about this